### DISCUSSION TITLE
"PltReolver: Symbol FcNameRegisterObjectTypes not found" [SOLVED]

### DISCUSSION THREAD

**Discussion by bozon781** [ptitSeb/box86#370] [Q&A] [answered]
TL;DR
I am trying to use my RPi4 (Running the latest version of 32bit Raspian OS) to run the linux version FullRecall http://fullrecall.com/download (flashcard / memorization software) and  have encountered "Error: PltReolver: Symbol FcNameRegisterObjectTypes not found" error.

Here are the instructions I followed verbatim - https://pimylifeup.com/raspberry-pi-x86/

I'm a beginner to Linux and (especially) to Raspberry Pi and have no idea what to do here. I've tried Googling but have not turned up anything actionable on "Error: PltReolver: Symbol FcNameRegisterObjectTypes not found". 

What does it mean and how do I fix this?

TIA for any assistance!

**Comment by ptitSeb** [comment]
You can't, that function handling needs to be added in box86 itself.

I have done it, you need to update box86 and try again.

**Reply by bozon781** [reply]
Thank you very much - that was super quick!

Unfortunately, the error now reads

Error: PltReolver: Symbol FcCharSetCount not found, cannot apply R_386_JMP_SLOT @0x8380380 (0x8055246) in ./fullrecall.bin

:( 

Would you mind looking at it again? I'd be happy to throw a few ducats your way to make FullRecall run!

**Comment by ptitSeb** [comment]
Ok, this one is added too. Hopefully there will not be a ton of missing symbols...

**Reply by bozon781** [reply]
Still no dice :( 
Error: PltReolver: Symbol FcFreeTypeCharIndex not found,  cannot apply R_386_JMP_SLOT @0x8380620 (0x8055cc6) in ./fullrecall.bin now :(

If this is any indication, I get a feeling there could be a lot of these  https://refspecs.linuxfoundation.org/fontconfig-2.6.0/x102.html

Could be that this might be just too much work on your end :(

**Comment by ptitSeb** [comment, marked as answer]
Yes, there are a few, but many are already wrapped. Look there https://github.com/ptitSeb/box86/blob/master/src/wrapped/wrappedfontconfig_private.h , everything not starting by `//` is done.

Also, I have pushed that new one.

**Reply by bozon781** [reply]
That did it! Superb! It works! 

Is there a way we can donate to Box86?

**Comment by ptitSeb** [comment]
Ah good ! 

(there is a "Sponsor" button on the box86 repo on github)

**Reply by jusn6935** [reply]
I am also getting a PltResolver Error. I wrote a small program in LiveCode and it works great on Ubuntu 32bit (ver16.04lts) but I get an error when I try to run in Raspbian 32bit (RPi 3B+). The error is as follows: Error: PltResolver: Symbol dladdr1(ver 14: dladdr1@GLIBC_2.3.3) not found, cannot apply R_386_JMP_SLOT 0x8dcd150 (0x8094466) in /home/pi/Desktop/32/Maw_Interface.
Thx a bunch

**Reply by JerBjer** [reply]
Hi ptitSeb! I think I have a similar issue. When trying to launch steam I get this error: 
Error: PltReolver: Symbol __dprintf_chk(__dprintf_chk@GLIBC_2.8) not found, cannot apply R_386_JMP_SLOT @0xb6c1ea24 (0xb664d516) in /home/pi/.local/share/Steam/ubuntu12_32/steam
Does this symbol also need to be added? 

Thanks!

**Reply by ptitSeb** [reply]
> Hi ptitSeb! I think I have a similar issue. When trying to launch steam I get this error: Error: PltReolver: Symbol __dprintf_chk(__dprintf_chk@GLIBC_2.8) not found, cannot apply R_386_JMP_SLOT @0xb6c1ea24 (0xb664d516) in /home/pi/.local/share/Steam/ubuntu12_32/steam Does this symbol also need to be added?
> 
> Thanks!

I already added that symbol with commit e31e5c1a5b254404287afdb0143a94b8ff7fcd56 (21st of january). Is your box86 up-to-date?

**Reply by ptitSeb** [reply]
> I am also getting a PltResolver Error. I wrote a small program in LiveCode and it works great on Ubuntu 32bit (ver16.04lts) but I get an error when I try to run in Raspbian 32bit (RPi 3B+). The error is as follows: Error: PltResolver: Symbol dladdr1(ver 14: dladdr1@GLIBC_2.3.3) not found, cannot apply R_386_JMP_SLOT 0x8dcd150 (0x8094466) in /home/pi/Desktop/32/Maw_Interface. Thx a bunch

I'll try to add that symbol "soon". It's a tricky one...

**Reply by JerBjer** [reply]
> > Hi ptitSeb! I think I have a similar issue. When trying to launch steam I get this error: Error: PltReolver: Symbol __dprintf_chk(__dprintf_chk@GLIBC_2.8) not found, cannot apply R_386_JMP_SLOT @0xb6c1ea24 (0xb664d516) in /home/pi/.local/share/Steam/ubuntu12_32/steam Does this symbol also need to be added?
> > Thanks!
> 
> I already added that symbol with commit [e31e5c1](https://github.com/ptitSeb/box86/commit/e31e5c1a5b254404287afdb0143a94b8ff7fcd56) (21st of january). Is your box86 up-to-date?

Could've sworn I was on the latest build, but I was not. It is working now! Thank you!
